### PR TITLE
infra: parallelize no-exception javadoc github workflow

### PR DESCRIPTION
**Pull request by nmancus1**
From https://github.com/checkstyle/checkstyle/pull/10143#pullrequestreview-690624945:

> it would be better to split such big job (21 min) to multiple each execution separately, it will make it more quick by parallelism and easy to restart only single execution.

### CODE REVIEW

File: .github/workflows/no-exception-workflow.yml [hunk 1]
```diff
@@ -5,11 +5,68 @@ on:
...
+      with:
+        java-version: 8
+
+    - name: Install checkstyle
+      run: mvn -e --no-transfer-progress clean install -Pno-validations
+
+    - name: Install dependencies
+      run: sudo apt install groovy
+
+    - run: ./.ci/no-exception-test.sh no-exception-cassandra-storm-tapestry-javadoc
```
Comment (romani) [inline .github/workflows/no-exception-workflow.yml:44]: please make order of runs like:
`Set up JDK 8`
`Install dependencies`
`Checkout repository`
`Install checkstyle`
` ./.ci/no-exception-test.sh`

**Review by romani — CHANGES_REQUESTED** [review]
item:

File: .github/workflows/no-exception-workflow.yml (hunk 1 above)
Comment (nmancus1) [reply .github/workflows/no-exception-workflow.yml:44]: Done

**Review by romani — APPROVED** [review]
ok to merge

**Review by pbludov — APPROVED** [review]

**Pull request merged**